zilliqa/src/pool.rs

```diff
@@ -211,6 +213,7 @@ impl TransactionPool {
             // keeping the same nonce. So for those, it will always discard the new (identical)
             // one.
             if ReadyItem::from(existing_txn) >= ReadyItem::from(&txn) {
+                debug!("Received txn with the same nonce but lower gas. Txn hash: {:?}, from: {:?}, nonce: {:?}, gas_limit: {:?}", txn.hash, txn.signer, txn.tx.nonce(), txn.tx.gas_limit());
                 return false;
```

Comment:
I guess you wanted to log the gas price and not the gas limit. The higher gas price makes a transaction with the same nonce better, not the higher gas limit...